presto-hive/src/main/java/com/facebook/presto/hive/S3SelectRecordCursorProvider.java

```diff
@@ -0,0 +1,103 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive;
+
+import com.facebook.presto.hive.s3.HiveS3Config;
+import com.facebook.presto.hive.s3.PrestoS3ClientFactory;
+import com.facebook.presto.spi.ConnectorSession;
+import com.facebook.presto.spi.PrestoException;
+import com.facebook.presto.spi.RecordCursor;
+import com.facebook.presto.spi.predicate.TupleDomain;
+import com.facebook.presto.spi.type.TypeManager;
+import com.google.common.collect.ImmutableSet;
+import io.airlift.log.Logger;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.Path;
+import org.joda.time.DateTimeZone;
+
+import javax.inject.Inject;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Optional;
+import java.util.Properties;
+import java.util.Set;
+
+import static com.facebook.presto.hive.HiveErrorCode.HIVE_FILESYSTEM_ERROR;
+import static com.facebook.presto.hive.HiveUtil.getDeserializerClassName;
+import static java.util.Objects.requireNonNull;
+
+public class S3SelectRecordCursorProvider
+        implements HiveRecordCursorProvider
+{
+    private static final Logger LOG = Logger.get(S3SelectRecordCursorProvider.class);
+    private static final Set<String> CSV_SERDES = ImmutableSet.of("org.apache.hadoop.hive.serde2.lazy.LazySimpleSerDe");
+    private final HdfsEnvironment hdfsEnvironment;
+    private final HiveClientConfig clientConfig;
+    private final PrestoS3ClientFactory s3ClientFactory;
+
+    @Inject
+    public S3SelectRecordCursorProvider(
+            HdfsEnvironment hdfsEnvironment,
+            HiveClientConfig clientConfig,
+            PrestoS3ClientFactory s3ClientFactory)
+    {
+        this.hdfsEnvironment = requireNonNull(hdfsEnvironment, "hdfsEnvironment is null");
+        this.clientConfig = requireNonNull(clientConfig, "clientConfig is null");
+        this.s3ClientFactory = requireNonNull(s3ClientFactory, "s3ClientFactory is null");
+    }
+
+    @Override
+    public Optional<RecordCursor> createRecordCursor(
+            Configuration configuration,
+            ConnectorSession session,
+            Path path,
+            long start,
+            long length,
+            long fileSize,
+            Properties schema,
+            List<HiveColumnHandle> columns,
+            TupleDomain<HiveColumnHandle> effectivePredicate,
+            DateTimeZone hiveStorageTimeZone,
+            TypeManager typeManager)
+    {
+        boolean isS3SelectPushdownEnabled = configuration.getBoolean(HiveS3Config.S3_SELECT_PUSHDOWN_ENABLED, false);
```

Comment:
static import `S3_SELECT_PUSHDOWN_ENABLED`

Reply:
Sorry for missing these static imports over and over again. I think it would be best to add these and non-final local variables checks in the style guide validations during build. Would save lot of time for reviewers.

Reply:
@sameer2086 i like your idea! You can see/edit checkstyle rules that we use @ https://github.com/airlift/airbase/blob/master/airbase-policy/src/main/resources/checkstyle/airbase-checks.xml

Reply:
btw why are we passing our "native" `HiveS3Config` configuration thru Hive's `Configuration` object? @nezihyigitbasi is it intentional?

Reply:
I was not aware that HiveS3Config is meant for only public facing configurations and constants. It makes sense to me to move this out of the class. I was treating it as a generic high level config and constant class for defining shared constants. 

Is there a common class for constants in code to define this? Does being more closely related to S3Select related logic makes S3SelectRecordCursorProvider a better place for the config than HivePageSourceProvider?

Reply:
For now moving the config to S3SelectRecordCursorProvider. Will change if we agree on another solution.

Reply:
airlift/airbase#168